thorlcr/master/thmastermain.cpp

```diff
@@ -300,10 +305,17 @@ class CRegistryServer : public CSimpleInterface
             else
             {
                 Owned<INode> sender = _sender;
-                if (NotFound != connectedSlaves.find(sender))
+                StringBuffer workerEPStr;
+                sender->endpoint().getEndpointHostText(workerEPStr);
+
+                auto findFunc = [&workerEPStr](ConnectedWorkerDetail& t)
+                {
+                    return std::get<0>(t) == std::string(workerEPStr.str());
```

Comment:
better would be

streq(std::get<0>(t).c_str(), workerEPStr.str())

otherwise it will be creating string instances for each match.

Reply:
Slightly concerned that this will be O(N^2) in the number of slaves, but it should be quick enough (and on-e-off) to not matter.

Reply:
>otherwise it will be creating string instances for each match.

changed.
Left linear search, don't think worth changing.